### ISSUE TITLE
aligned allocation in c++17 dialect

### ISSUE THREAD

**Issue by squm** [opencv/opencv#15363]
> Headers/core/utility.hpp:581:7: error: aligned deallocation function of type 'void (void *, enum std::align_val_t) noexcept' is only available on iOS 11 or newer
> 
> class ParallelLoopBodyLambdaWrapper : public ParallelLoopBody
>       ^
> 
> Headers/core/utility.hpp:598:26: note: in implicit destructor for 'cv::ParallelLoopBodyLambdaWrapper' first required here
> 
>     parallel_for_(range, ParallelLoopBodyLambdaWrapper(functor), nstripes);
>                          ^
> 
> Headers/core/utility.hpp:581:7: note: if you supply your own aligned allocation functions, use -faligned-allocation to silence this diagnostic
> 
> class ParallelLoopBodyLambdaWrapper : public ParallelLoopBody

**Comment by rasha108** [comment]
resolved but no?

**Comment by fzyzcjy** [comment]
Hi I am having the same problem. Did you solve it?

**Comment by squm** [comment]
> Hi I am having the same problem. Did you solve it?

no, only by switching to std 11. Looks like this isn't fixed here either https://github.com/Microsoft/cpprestsdk/issues/1000

**Comment by squm** [comment]
another option is to open Build Phases → Compile Sources and add to _Compiler Flags_ `-fno-aligned-allocation` to the files which include opencv header. You can select multiple files at once and filter with search

**Comment by squm** [comment]
however I hope opencv maintainers will tell how their software will react to that ? Perhaps they have a reason to use it, will it result in undefined behavior ?
